ui/src/components/services/CreateServiceForm.js

```diff
@@ -63,92 +124,211 @@ export default class CreateServiceForm extends React.Component {
+          <Form.Input name="Args" label="Args" placeholder="Default" />
+        </Form.Group>
 
-				<Divider hidden />
+        <Divider hidden />
 
         <Form.Group widths="equal">
-					<Form.Field name="Mode" label="Mode" onChange={(e) => this.handleChange('Mode', e)} control="select" required>
-						<option value=""></option>
-						<option value="Replicated">Replicated</option>
-						<option value="Global">Global</option>
-					</Form.Field>
-          <Form.Input name="Replicas" label="Replicas" type="number" disabled={Mode !== 'Replicated'} required={Mode === 'Replicated'} placeholder={1} />
+          <Form.Select name="Mode" label="Mode" onChange={(e, data) => this.handleChange(data.name, data.value)} options={modes} placeholder="Mode" required />
+          <Form.Input name="Replicas" label="Replicas" type="number" disabled={Mode !== "Replicated"} required={Mode === "Replicated"} placeholder={1} />
         </Form.Group>
 
-				<Divider hidden />
-
-				<Form.Group widths="equal">
-					<Form.Input name="User" label="User" placeholder="nobody" />
-					<Form.Input name="Dir" label="Working Directory" placeholder="/usr/src/app" />
-					<Form.Input name="Groups" label="Groups" placeholder="group1 group2" />
-				</Form.Group>
-
-				{/* These don't seem to do anything at the moment 
-				<Form.Field>
-					<label>Stream Options</label>
-					<Form.Group inline>
-						<Form.Checkbox label='Attach TTY' name='TTY' />
-						<Form.Checkbox label='Open stdin' name='OpenStdin' />
-					</Form.Group>
-				</Form.Field>
-				*/}
-
-				<Divider hidden />
+        <Divider hidden />
+
+        <Form.Group widths="equal">
+          <Form.Input name="User" label="User" placeholder="nobody" />
+          <Form.Input name="Dir" label="Working Directory" placeholder="/usr/src/app" />
+          <Form.Input name="Groups" label="Groups" placeholder="group1 group2" />
+        </Form.Group>
+
+        <Divider />
+
+        <List>
+          {this.state.mountInputKeys.map((key, i) => {
+            return (
+              <List.Item key={key}>
+                <Form.Group widths="equal">
+                  <Form.Select name={"Mounts.MountType-" + i} label={i === 0 && "Type"} options={mountTypes} placeholder="Type" required />
+                  <Form.Input name={"Mounts.Source-" + i} label={i === 0 && "Source"} placeholder="volumename or /host/path" required />
+                  <Form.Input name={"Mounts.Target-" + i} label={i === 0 && "Target"} placeholder="/container/path" required />
+                  <Form.Input name={"Mounts.VolumeDriver-" + i} label={i === 0 && "Volume Driver"} placeholder="local" required />
+                  <Form.Select name={"Mounts.ReadOnly-" + i} label={i === 0 && "Read Only"} options={readOnlyOptions} placeholder="Read Only" required />
+                  <Form.Button type="button" tabIndex={i} width={4} icon="minus" label={i === 0 && "Remove"} basic onClick={this.removeMount}/>
```

Comment:
A splash of colour on the add/remove buttons might help call out these buttons :)  Can add that in a future PR though if you were interested in adding the attach network UX.